paasta_tools/kubernetes_tools.py

```diff
@@ -1134,6 +1139,28 @@ def get_pod_template_spec(
             if "hpa" in annotations:
                 annotations["hpa"] = json.dumps(annotations["hpa"])
 
+        pod_spec_kwargs = dict(
+            service_account_name=self.get_kubernetes_service_account_name(),
+            containers=self.get_kubernetes_containers(
+                docker_volumes=docker_volumes,
+                aws_ebs_volumes=self.get_aws_ebs_volumes(),
+                system_paasta_config=system_paasta_config,
+                service_namespace_config=service_namespace_config,
+            ),
+            share_process_namespace=True,
+            node_selector=self.get_node_selector(),
+            restart_policy="Always",
+            volumes=self.get_pod_volumes(
+                docker_volumes=docker_volumes,
+                aws_ebs_volumes=self.get_aws_ebs_volumes(),
+            ),
+        )
+        # need to check if there are node selectors/affinities. if there are none
+        # and we create an empty affinity object, k8s will deselect all nodes.
+        node_affinity = self.get_node_affinity()
+        if node_affinity is not None:
+            pod_spec_kwargs["affinity"] = V1Affinity(node_affinity=node_affinity)
```

Comment:
What if the method was just `get_affinity` and returned an Optional[V1Affinity]? I think it would make the code simpler.